src/Expr/ClosureExpressionVisitor.php

```diff
@@ -178,7 +178,7 @@ public function walkCompositeExpression(CompositeExpression $expr): Closure
     /** @param callable[] $expressions */
     private function andExpressions(array $expressions): Closure
     {
-        return static fn ($object): bool => array_all(
+        return static fn (object $object): bool => array_all(
```

Comment:
I didn't use `object|array` here though. Running the ORM test suite with that type hint causes no issues.